### PR TITLE
Vignette data set

### PR DESCRIPTION
**Pull request by PavelBal**
Added a vignette on how to work with `DataSet` objects and how to load data from excel.

@IndrajeetPatil Maybe this can be later integrated into the vignette on `DataCombined` or mentioned it there?

### CODE REVIEW

File: vignettes/observed-data.Rmd [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+`ospsuite` offers a concept of storing and processing numerical x-y data in a
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:18]:
```suggestion
`{ospsuite}` offers a concept of storing and processing numerical x-y data in a 
```

**Comment by PavelBal** [comment]
@msevestre could you please add @svavil to the team so I can request his review?

File: vignettes/observed-data.Rmd [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  comment = "#>"
+)
+```
+
+`ospsuite` offers a concept of storing and processing numerical x-y data in a 
+unified format by implementing a `DataSet` class. `DataSet` objects standardize 
+handling of (observed) data coming from different sources, such as excel files 
+using the data importer functionality of the OSPS, loaded from *.pkml, or manually 
+created. This vignette gives an overview of the options to create `DataSet` objects
+ and combine them into grouped data sets using the `DataCombined` class
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:23]:
```suggestion
 and combine them into grouped data sets using the `DataCombined` class.
```

File: vignettes/observed-data.Rmd [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+## DataSet
+
+A `DataSet` objects stores numerical data pairs - typically the time
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:27]:
```suggestion
A `DataSet` object stores numerical data pairs - typically time 
```

File: vignettes/observed-data.Rmd [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+handling of (observed) data coming from different sources, such as excel files 
+using the data importer functionality of the OSPS, loaded from *.pkml, or manually 
+created. This vignette gives an overview of the options to create `DataSet` objects
+ and combine them into grouped data sets using the `DataCombined` class
+
+## DataSet
+
+A `DataSet` objects stores numerical data pairs - typically the time 
+as x values and measurement as y values - and optionally the (statistical) error of 
+the measurement. All values have a *dimension*  and a *unit* (see [Dimensions and Units](unit-conversion.html)
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:29]:
```suggestion
as x values and measurement as y values - and optionally the measurement error . All values have a *dimension*  and a *unit* (see [Dimensions and Units](unit-conversion.html) 
```

File: vignettes/observed-data.Rmd [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+```
+
+After creation, the `DataSet` does not hold any data. The default dimension and 
+unit for the x values is `Time` and `h`, respectively. The default dimension and 
+unit for the y values is `Concentration (mass)` and `mg/l`, respectively. The 
+dimension of the error values always corresponds to the dimension of the y values, 
+though the units may differ.
+
+Setting numerical values (or overwriting current values) is performed by the 
+`setValues()` method:
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:48]:
```suggestion
`$setValues()` method:
```

File: vignettes/observed-data.Rmd [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+# Change the units of y values and error values - they are now different!
+dataSet$yUnit <- ospUnits$Amount$mol
+dataSet$yErrorUnit <- ospUnits$Amount$pmol
+
+print(dataSet)
+```
+
+Two types of error values are supported - arithmetic error (default) and geometric 
+error, the latter being given in fraction. The user can change error type:
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:88]:
```suggestion
error, the latter being given in fraction. The user can change the error type:
```

File: vignettes/observed-data.Rmd [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+dataSet$yErrorUnit
+
+# Changing error type to arithmetic will set the dimension and unit of the error 
+# to the same dimension and unit as the y values
+dataSet$yErrorType <- DataErrorType$ArithmeticStdDev
+
+print(dataSet)
+```
+
+A `DataSet` can store any kind of text meta data as meta data name - values pairs
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:106]:
```suggestion
A `DataSet` can store any kind of text meta data as name-values pairs
```

File: vignettes/observed-data.Rmd [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+                    value = "Aciclovir")
+dataSet$addMetaData(name = "Organ",
+                    value = "Muscle")
+
+# Print meta data of the DataSet
+print(dataSet$metaData)
+```
+ 
+A `DataSet` or multiple `DataSet`s  can be converted to `data.frame` (or `tibble`) 
+to be processed in establised R workflows:
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:121]:
```suggestion
to be processed in downstream analysis and visualization workflows:
```

File: vignettes/observed-data.Rmd [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+`ospsuite` offers a concept of storing and processing numerical x-y data in a 
+unified format by implementing a `DataSet` class. `DataSet` objects standardize 
+handling of (observed) data coming from different sources, such as excel files 
+using the data importer functionality of the OSPS, loaded from *.pkml, or manually
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:21]:
```suggestion
using the data importer functionality of the OSPS, loaded from `*.pkml`, or manually 
```

File: vignettes/observed-data.Rmd [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+# Load a data set from PKML
+filePath <- system.file("extdata", "ObsDataAciclovir_1.pkml", package = "ospsuite")
+
+dataSet <- loadDataSetFromPKML(filePath = filePath)
+
+print(dataSet)
+```
+
+Another (and probably the most important) way to create `DataSet` objects is to 
+import data from excel files. The function `loadDataSetsFromExcel()` utilizes the
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:153]:
```suggestion
Another (and probably the most important) way to create `DataSet` objects is by 
importing data from excel files. The function `loadDataSetsFromExcel()` utilizes the 
```

File: vignettes/observed-data.Rmd [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+# Load data
+dataSets <- loadDataSetsFromExcel(xlsFilePath = excelFilePath,
+                      importerConfigurationOrPath = importerConfiguration_guessed)
+
+```
+
+Currently, `DataImporterConfiguration` created from scratch or for a specific 
+data sheet does not support all features of importer configuration, such as 
+specifying the column containing the molecular weight of the measured molecule 
+or the LLOQ values. It is therefore recommended to use importer configurations
```
Comment (IndrajeetPatil) [inline vignettes/observed-data.Rmd:216]:
```suggestion
or the LLOQ values. It is therefore recommended that you use importer configurations 
```

**Review by IndrajeetPatil — APPROVED** [review]
This is excellent! No big changes or suggestions from my end.

And, yes, we can link to this vignette from the `DataCombined` class vignette.

**Comment by IndrajeetPatil** [comment]
Yepp, merged #1071 because it is needed for this PR.

**Pull request merged**